### PR TITLE
Init NetBox Operator Helm chart

### PR DESCRIPTION
**Pull request by LeoColomb**
This PR implements a Helm chart for the [NetBox Operator](https://github.com/netbox-community/netbox-operator).

Resolves https://github.com/netbox-community/netbox-operator/issues/105

```[tasklist]
- [x] Templates & Rendering
- [x] Integration tests
- [x] READMEs and docs
```

### CODE REVIEW

File: charts/netbox-operator/values.yaml [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+  ##
+  pullSecrets: []
+
+## @section NetBox Operator Configuration parameters
+
+## NetBox Operator settings based on environment variables
+caCert: ""
+host: ""
+https: false
+restorationHashFieldName: "netboxOperatorRestorationHash"
```
Comment (jstudler) [inline charts/netbox-operator/values.yaml:88]: We already default this in the Operator. Maybe we can comment this out in the values.yaml?

Comment (LeoColomb) [reply charts/netbox-operator/values.yaml:88]:
```suggestion
restorationHashFieldName: ""
```

**Comment by LeoColomb** [comment]
Thanks for your review, @jstudler!
Indeed, just switched to an empty default so it can use the upstream value.

While I think I'm mostly done with the chart, I struggle to set up a correct CI environment.
Having a NetBox automatically configured with `restorationHashFieldName` custom field seems difficult to automate.
Do you have an instance available for test somewhere? If not, do you have a way to automate?

**Comment by jstudler** [comment]
For automated deployment on a local kind cluster you can use `make create-kind && make deploy-kind` from https://github.com/netbox-community/netbox-operator/blob/main/Makefile

We currently just load a SQL file to the database (see https://github.com/netbox-community/netbox-operator/blob/main/kind/load-data-job/local-demo-data.sql#L3)

File: charts/netbox-operator/ci/default-values.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+host: demo.netbox.dev
+https: true
+auth:
+  apiToken: "55570df95c0c850ada67a4185228732aaffd4964"
```
Comment (RangerRick) [inline charts/netbox-operator/ci/default-values.yaml:4]: should there be a hardcoded token in here?

Comment (LeoColomb) [reply charts/netbox-operator/ci/default-values.yaml:4]: @RangerRick No, but I'm waiting for an official instance provided by NetBox cloud 😇
This one is used for the demo instance, so it's all fine and probably deleted already.

Comment (LeoColomb) [reply charts/netbox-operator/ci/default-values.yaml]: Fixed

**Review by jstudler — APPROVED** [review]
Just tested this, looks good to me!

**Review by RangerRick — APPROVED** [review]
let's do it

**Pull request merged**